src/utils.jl

```diff
@@ -0,0 +1,16 @@
+"""
+    submat(A::AbstractPDMat, inds)
+
+Select a square submatrix of an `AbstractPDMat` specified by `inds`.
+Returns a `AbstractPDMat` of the same type.
+"""
+submat(A::PDMat, inds) = PDMat(A[inds, inds])
+submat(A::PSDMat, inds) = PSDMat(A[inds, inds])
+submat(A::PDiagMat, inds) = PDiagMat(A.diag[inds])
+submat(A::PDSparseMat, inds) = PDSparseMat(sparse(A[inds, inds]))
+submat(A::WoodburyPDMat, inds) = WoodburyPDMat(A.A[inds, :], A.D, Diagonal(A.S.diag[inds]))
+function submat(A::ScalMat, inds)
+    # inds should not be able to "access" non-existent indices
```

Comment:
this bound check is not required for other types of PDMat?

Reply:
no because for the ScalMat we're not really indexing into it, we're just returning a smaller ScalMat. But that means there's nothing stopping you from asking for the 1_000_000th index if you wanted. So this check makes sure you can't do that.